### PR TITLE
feat(tree-view): add multiselect support

### PR DESCRIPTION
**Pull request by angelcabo**
Adds multi-select support to the tree-view component.

## Description

Allows the use of shiftKey & metaKey while clicking tree-view-items to allow multi-selecting items within a root-level tree-view component.

## Related Issue

https://github.com/adobe/spectrum-web-components/issues/1222

## Motivation and Context

This change adds similar multi-select functionality highlighted on the spectrum-css site under the standard tree-view section: https://opensource.adobe.com/spectrum-css/treeview.html#standard

## How Has This Been Tested?

Yes. Tested manually with storybook and automated tests.

## Screenshots (if appropriate):

Screen Shot 2021-04-29 at 11 16 04 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/tree-view/src/TreeView.ts [hunk 1]
```diff
@@ -33,50 +34,60 @@ export class TreeView extends SpectrumElement {
...
+    public selectedChildren: TreeViewItem[] = [];
 
     @property({ type: Boolean, reflect: true })
     public standalone = false;
 
     @property({ type: Boolean, reflect: true })
     public quiet = false;
 
+    @property({ type: Boolean, reflect: true, attribute: 'allow-multiple' })
+    public allowMultiple = false;
```
[the remark this answers is not in the document]
Comment (Westbrook) [reply packages/tree-view/src/TreeView.ts:46]: What we decide here should manage the `aria-multiselectable="true"` attribute for us.

File: packages/tree-view/src/TreeViewItem.ts [hunk 2]
```diff
@@ -94,9 +94,19 @@ export class TreeViewItem extends ObserveSlotPresence(
         this.open = !this.open;
     }
 
-    public toggleSelected(event: Event): void {
+    public toggleSelected(event: MouseEvent): void {
         event.preventDefault();
         this.selected = !this.selected;
+        const multiselect = event.shiftKey || event.metaKey;
+        const applyDefault = this.dispatchEvent(
```
Comment (Westbrook) [inline packages/tree-view/src/TreeViewItem.ts:101]: This currently dispatches `selected` and `deselected` events as a reaction to the change to `this.selected` as well: https://github.com/adobe/spectrum-web-components/pull/1423/files#diff-b84c849cef36d2ec108a54310d7b0c4b3989e7bc837147666dad0e5a4ec5e50fR157 should be one of the other, IMO.

File: packages/tree-view/src/TreeViewItem.ts [hunk 3]
```diff
@@ -94,9 +94,19 @@ export class TreeViewItem extends ObserveSlotPresence(
...
+            new CustomEvent('toggled', {
+                bubbles: true,
+                composed: true,
+                cancelable: true,
+                detail: { multiselect },
+            })
+        );
+        if (!applyDefault) this.selected = !this.selected;
```
[the remark this answers is not in the document]
Comment (Westbrook) [reply packages/tree-view/src/TreeViewItem.ts:109]: I missed it but selection should manage `aria-selected="true|false"`, but this should also only be used when a selection can be made...

**Review by Westbrook — COMMENTED** [review]
Thanks for starting to take a look at this! I'd love to hear more of how you team might be leveraging this feature to ensure we get it structured correctly for everyone to consume.

The accessibility review here is based off of a cursory reading of https://w3c.github.io/aria-practices/#TreeView and likely has a lot more things even beyond what I've notes already.

With multi-select included, I'd want to make sure that `sp-tree-view` surfaces a `selectAll` and `deselectAll` functionality in some way. Maybe public methods, but I'd be happy to take suggestions.

Let me know if there is anything I can do to support getting the next revision of this together.

**Comment by angelcabo** [comment]
Closing this PR in favor of https://github.com/adobe/spectrum-web-components/pull/1478

**Pull request closed without merging**